docs/tutorials/09_Protein_Folding.ipynb

```diff
@@ -0,0 +1,824 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2b28a00e",
+   "metadata": {},
+   "source": [
+    "### Side Chains"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b99c81a8",
+   "metadata": {},
+   "source": [
+    "Beyond the main chain of the protein there may be aminoacids attached to the residues of the main chain. Our model allows for side chains of the maximum length of one. Elongated side chains would require the introduction of additional penalty terms which are still under development. In this example we do not consider any side chains to keep the real structure of the neuropeptide. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "f0cea914",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "side_chains = [\"\"]*7"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "203e288b",
+   "metadata": {},
+   "source": [
+    "### Interaction between Aminoacids"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72ff748d",
+   "metadata": {},
+   "source": [
+    "For the description of inter-residue contacts for proteins we use knowledge-based (statistical) potentials derived using quasi-chemical approximation. The potentials used here are introduced by Miyazawa, S. and Jernigan, R. L. in [5]. \n",
+    "\n",
+    "Beyond this model we also allow for random contact maps (interactions) that provide a random interaction map. One can also introduce a custom interaction map that enhances certain configurations of the protein (e.g. alpha helix, beta sheet etc). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cca5df13",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "random_interaction = RandomInteraction()\n",
+    "mj_interaction = MiyazawaJerniganInteraction()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb5c3ec8",
+   "metadata": {},
+   "source": [
+    "### Physical Constraints"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5adb4c73",
+   "metadata": {},
+   "source": [
+    "To ensure that all physical constraints are respected we introduce penalty functions. The different penalty terms used are: \n",
+    "\n",
+    "- penalty_chiral: A penalty parameter used to impose the right chirality.\n",
+    "\n",
+    "- penalty_back: A penalty parameter used to penalize turns along the same axis. This term is used to eliminate sequences where the same axis is chosen twice in a row. In this way we do not allow for a chain to fold back into itself.\n",
+    "\n",
+    "- penalty_1: A penalty parameter used to penalize local overlap between beads within a nearest neighbor contact."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "d96678c3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "penalty_back = 10\n",
+    "penalty_chiral = 10\n",
+    "penalty_1 = 10\n",
+    "\n",
+    "penalty_terms = PenaltyParameters(penalty_chiral, penalty_back, penalty_1)"
```

Comment:
We could potentially make use of default values which are also 10. But perhaps it is better to be explicit in the tutorial...
```suggestion
    "\n",
    "penalty_terms = PenaltyParameters()"
```

Reply:
I think the tutorial should be explicit here :+1: